### PR TITLE
Fix site-to-site VPN feature

### PR DESCRIPTION
**Pull request by remibergsma**
This is work done together with @jayapalu on fixing the site2site VPN. The first part was done in PR #690 by @jayapalu. On top of that, some other fixes were needed and those are added in this PR. It made sense to make a new PR which includes all fixes so we can actually test it. 

The original PR #690 is already merged into this one, so can be closed. Since the commit ids are kept the same, merging this will close both.

I closely compared the 4.4/4.5 implementation with the new 4.6 one. I did not only make it work, but also added some security improvements (some of which were also in 4.4/4.5). I noticed the pre shared key was being logged, so removed that as well.

This is how I tested and verified it:
https://github.com/schubergphilis/MCT-shared/tree/master/helper_scripts/cloudstack/vpn_tests
When I have some time available, I'll write a Marvin test for it that we can include in the repo.

It now works(tm) with one manual step due to CLOUDSTACK-8685: 
We need a default gateway before site-to-site VPN will actually work. It will connect, but not forward packets. The reason for this, is due to the iptables setup. VM1 has router1 as gateway, but router1 does not know the route to VM2 so it will give up. With a default gateway, the packets are about to be forwarded to the default gateway but when they reach eth1 the public nic, iptables kicks in, does some magic and forwards it through the ipsec tunnel. So, you need a default gw set to upstream.

Workaround for now is setting the route manually:
``route add default gw 1.2.3.4``  or  ``ip route add default via 1.2.3.4``

In other words, we need to fix CLOUDSTACK-8685 soon, too.

Thanks to @snuf @jayapalu!
 
@jayapalu @snuf could you please review this?

### CODE REVIEW

File: systemvm/patches/debian/config/opt/cloud/bin/cs/CsFile.py [hunk 1]
```diff
@@ -114,7 +114,10 @@ def greplace(self, search, replace):
 
     def search(self, search, replace):
         found = False
-        logging.debug("Searching for %s and replacing with %s" % (search, replace))
+        replace_filtered = replace
+        if re.search("PSK \"", replace):
+            replace_filtered = re.sub(r'".*"', '"****"', replace)
+        logging.debug("Searching for %s and replacing with %s" % (search, replace_filtered))
```
Comment (DaanHoogland) [inline systemvm/patches/debian/config/opt/cloud/bin/cs/CsFile.py]: "Searching for %s and replacing the value following it with %s" would be closer to the truth

**Comment by DaanHoogland** [comment]
no furhter comments, LGTM

**Comment by jayapal** [comment]
Guys wrong tag.

Myuser name is jayapal and actual commiter was jayapalu

**Comment by remibergsma** [comment]
@jayapal Sorry for bothering you, just updated the tag. Thanks for letting me know!

**Comment by jayapalu** [comment]
The iptables rules and vpn config changes are looking good.
LGTM

**Comment by snuf** [comment]
LGTM

**Comment by jayapal** [comment]
No problem

Best,
Jayapal
Mobile, xcuse typoerror.
